agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -519,9 +515,34 @@ func (fv *FSxWindowsFileServerResource) retrieveFileSystemDNSName(fileSystemId s
 	return nil
 }
 
+var mountLock sync.Mutex
+var DriveLetterAvailable = utils.IsAvailableDriveLetter
 var execCommand = exec.Command
 
-func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, localPathArg string, username string, password string) error {
+func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, username string, password string) error {
+	mountLock.Lock()
```

Comment:
Can we extract this code into its own function? That way we don't need a new `Mutex`, plus the critical section locked by the mutex is reduced to the absolute minimum (the new mutex in this PR is now serializing all the powershell executions below as well, potentially making task creation slower)

e.g.

```golang
func (fv *FSxWindowsFileServerResource) defineHostPath() error {
	fv.lock.Lock()
	defer fv.lock.Lock()

	hostPath, err := utils.FindUnusedDriveLetter()
	if err != nil {
		return err
	}

	fv.VolumeConfig.HostPath = hostPath

	// formatting to keep powershell happy
	localPathArg := fmt.Sprintf("-LocalPath \"%s\"", strings.Trim(hostPath, `\`))

	// remove mount if localPath is not available
	// handling case where agent crashes after mount and before resource status is updated
	if !(DriveLetterAvailable(hostPath)) {
		err = fv.removeHostMount(localPathArg)
		if err != nil {
			seelog.Errorf("Failed to remove existing fsxwindowsfileserver resource mount on the container instance: %v", err)
			fv.setTerminalReason(err.Error())
			return err
		}
	}
}

func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, username string, password string) error {

	fv.defineHostPath()	
	.
	.
	.
```

Reply:
Nvm, after talking offline, it seems that the letter becomes unavailable only after executing the mounts, so the lock seems necessary after all.